zaza/model.py

```diff
@@ -1420,8 +1420,20 @@ def f(x):
     async def _unit_status():
         app = unit_name.split("/")[0]
         model_status = await async_get_status()
-        v = model_status.applications[app]['units'][unit_name][
-            'workload-status']['status']
+        if model_status.applications[app]['units']:
+            v = model_status.applications[app]['units'][unit_name][
+                'workload-status']['status']
+        else:
+            # For when the unit is a subordinate we need to get it's
+            # leader, and then get the status for the subordinate from it's
+            # unit
+            lead_app_name = model_status.applications[app]['subordinate-to'][0]
+            units = model_status.applications[lead_app_name]['units']
+            for unit in units:
+                if units[unit]['subordinates'][unit_name]:
```

Comment:
This can throw a KeyError, it is safer to do `if units[unit]['subordinates'].get(unit_name, None):` as that way you'll get the None in the falsey case, rather than getting a KeyError

Reply:
updated as suggested